### ISSUE TITLE
[1.19] cluster-reset command not completing

### ISSUE THREAD

**Issue by rancher-max** [rancher/rke2#1574]
**Environmental Info:**
RKE2 Version: 

v1.19.13+rke2r1 and v1.19.13-rc1+rke2r2

Node(s) CPU architecture, OS, and Version: 

SLES15 SP2 `Linux ip-172-31-13-142 5.3.18-24.37-default #1 SMP Wed Nov 4 09:38:41 UTC 2020 (c145e08) x86_64 x86_64 x86_64 GNU/Linux`

Cluster Configuration:

3 servers 1 agent

**Describe the bug:**

running `rke2 server --cluster-reset` just hangs in an endless loop of the following logs:
```
INFO[0270] Waiting for etcd server to become available  
INFO[0277] Cluster-Http-Server 2021/08/09 03:32:01 http: TLS handshake error from 172.31.5.133:43540: EOF 
{"level":"warn","ts":"2021-08-09T03:32:05.053Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
ERRO[0280] Failed to check local etcd status for learner management: context deadline exceeded 
{"level":"warn","ts":"2021-08-09T03:32:05.076Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
INFO[0280] Failed to test data store connection: context deadline exceeded 
^C{"level":"warn","ts":"2021-08-09T03:32:05.650Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
ERRO[0281] failed to check etcd status for snapshot: context canceled 
WARN[0281] Unable to watch for tunnel endpoints: Get "https://127.0.0.1:6443/api/v1/namespaces/default/endpoints?fieldSelector=metadata.name%3Dkubernetes&resourceVersion=0&watch=true": context canceled 
{"level":"warn","ts":"2021-08-09T03:32:05.664Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = Canceled desc = latest balancer error: connection error: desc = \"transport: authentication handshake failed: read tcp 127.0.0.1:45576->127.0.0.1:2379: read: connection reset by peer\""}
ERRO[0281] failed to check etcd status for snapshot: context canceled 
INFO[0281] Proxy done                                    err="context canceled" url="wss://127.0.0.1:9345/v1-rke2/connect"
{"level":"warn","ts":"2021-08-09T03:32:05.664Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"passthrough:///https://127.0.0.1:2379","attempt":0,"error":"rpc error: code = Canceled desc = latest balancer error: connection error: desc = \"transport: authentication handshake failed: read tcp 127.0.0.1:45944->127.0.0.1:2379: read: connection reset by peer\""}
ERRO[0281] failed to check etcd status for snapshot: context canceled 
```

**Steps To Reproduce:**

- Installed RKE2 v1.19.13-rc1+rke2r2 with 3 servers and 1 agent
- Stop/terminate the agent and 2 of the server machines (cluster will lose quorum) 
- On the remaining server, run cluster reset commands from the docs:
```
systemctl stop rke2-server
rke2 server --cluster-reset
```

**Expected behavior:**

cluster should reset properly and the process should end after a little bit with: `INFO[0085] Etcd is running, restart without --cluster-reset flag now. Backup and delete ${datadir}/server/db on each peer etcd server and rejoin the nodes`

**Actual behavior:**

This never happens and it is in an endless loop. Cannot reset the cluster after quorum loss.

**Additional context / logs:**

This also fails after trying to run `rke2-killall.sh` script before running the reset command, instead of only stopping the server.

**Comment by brandond** [comment]
This appears to be a duplicate of https://github.com/rancher/rke2/issues/1052 - I'm not sure if the fix for that was backported to 1.19; it's possible we didn't identify it as affected, or perhaps it was not previously affected.

**Comment by dereknola** [comment]
Validated on `v1.19.15-rc2+rke2r1` release:

Started up 3 rke2 server nodes, waited till quorum was reached 
```
root@node-cluster-init-buster64:/home/vagrant# kubectl get nodes
NAME                         STATUS   ROLES         AGE    VERSION
node-cluster-init-buster64   Ready    etcd,master   6m3s   v1.19.15+rke2r1
node-etcd-buster64           Ready    etcd,master   119s   v1.19.15+rke2r1
node-worker-buster64         Ready    etcd,master   77s    v1.19.15+rke2r1
```

Killed two of the servers
On the last server, see `FATA[0579] leaderelection lost for rke2 `, then ran
```
systemctl stop rke2-server
rke2 server --cluster-reset
...
INFO[0024] Etcd is running, restart without --cluster-reset flag now. Backup and delete ${datadir}/server/db on each peer etcd server and rejoin the nodes 
```
No endless loop, command returns correctly after a wait.
